@here/harp-mapview/lib/MapView.ts

```diff
@@ -1841,11 +1840,8 @@ export class MapView extends THREE.EventDispatcher {
      * @param textElements Array of [[TextElement]] to be added.
      */
     addOverlayText(textElements: TextElement[]): void {
-        if (this.m_overlayTextElements !== undefined) {
-            this.m_overlayTextElements = this.m_overlayTextElements.concat(textElements);
-        }
         this.createTextRendererIfNeeded();
-        this.m_textElementsRenderer!.placeAllTileLabels();
+        this.m_textElementsRenderer!.addOverlayText(textElements);
```

Comment:
agreed. But in my next changes I'm actually getting rid of the lazy construction of the TextElementsRenderer as discussed with Nino. I'm instead initializing the renderer lazily but the instance is created on MapView's constructor.
Since there's many places in map view doing **this.m_textElementsRenderer!** without an assert, and I'll merge the follow-up changes soon enough, it doesn't make much sense to do this here.